jenkins/scripts/utils.sh

```diff
@@ -60,15 +60,11 @@ vm_healthy()
     KEY="${2:?}"
     SERVER="${3:?}"
 
-    cloud_init_status=$(ssh -o ConnectTimeout=2 -o StrictHostKeyChecking=no \
+    if ssh -o ConnectTimeout=2 -o StrictHostKeyChecking=no \
```

Comment:
Noticed that `cloud-init` might return 1 for fatal, or 2 for recoverable error. This converts both to fatals. Just making sure this is what we want?

Reply:
```console
19:35:58  SSH connection to host[10.100.10.180] is up.
19:35:58  Warning: Permanently added '10.100.10.180' (ED25519) to the list of known hosts.
19:36:05  errors: []
19:36:05  recoverable_errors:
19:36:05  There was a cloud-init error:
19:36:05  ...........................
19:36:05  status: done
19:36:05  extended_status: degraded done
19:36:05  boot_status_code: enabled-by-generator
19:36:05  last_update: Thu, 29 Feb 2024 17:36:04 +0000
19:36:05  detail:
19:36:05  DataSourceOpenStackLocal [net,ver=2]
19:36:05  errors: []
19:36:05  recoverable_errors:
19:36:05  WARNING:
19:36:05  	- Invalid network-config provided: config.0.mtu: None is not of type 'integer' config.1.address: '1.1.1.1' is not of type 'array' config.2.address: '8.8.8.8' is not of type 'array'
19:36:05  Server is unhealthy. Giving up.
```
This would actually be the exact case here. The error is recoverable warning, so the return value would be 2, but this check would treat it as fatal.

Reply:
Good catch @tuminoid

Reply:
Oh I had missed that actually! I think we are fine with it this way. Checking the logs it looks like it didn't count the warning as a recoverable error:
```
10:44:57  status: done
10:44:57  extended_status: degraded done
10:44:57  boot_status_code: enabled-by-generator
10:44:57  last_update: Fri, 01 Mar 2024 08:44:57 +0000
10:44:57  detail:
10:44:57  DataSourceOpenStackLocal [net,ver=2]
10:44:57  errors: []
10:44:57  recoverable_errors:
10:44:57  WARNING:
10:44:57  	- Invalid network-config provided: config.0.mtu: None is not of type 'integer' config.1.address: '1.1.1.1' is not of type 'array' config.2.address: '8.8.8.8' is not of type 'array'
```

I confirmed also by sshing to the VM:

```
metal3ci@ci-test-vm-20240301084413-hxzq:~$ cloud-init status --long 
status: done
extended_status: degraded done
boot_status_code: enabled-by-generator
last_update: Fri, 01 Mar 2024 08:44:57 +0000
detail:
DataSourceOpenStackLocal [net,ver=2]
errors: []
recoverable_errors:
WARNING:
	- Invalid network-config provided: config.0.mtu: None is not of type 'integer' config.1.address: '1.1.1.1' is not of type 'array' config.2.address: '8.8.8.8' is not of type 'array'
metal3ci@ci-test-vm-20240301084413-hxzq:~$ echo $?
0

```

Reply:
I guess we should report to Cleura also that we get invalid network-config...

Reply:
It seems Ubuntu 22.04 does not have error 2 anymore. It either fails or succeeds, and recoverable error is success.
However, Ubuntu 20.04 and Centos do have error 2 still, which this will turn into fatal.